src/app/_hooks/useSearch.ts

```diff
@@ -0,0 +1,36 @@
+import { useMemo } from 'react'
+
+import { NextServerSearchParams } from '@/types/searchParams'
+import { SearchableByTitle } from '@/types/searchTypes'
+
+import { SEARCH_KEY } from '@/constants/searchParams'
+
+function validateSearchQuery(searchQuery: NextServerSearchParams['search']) {
+  if (!searchQuery) {
+    return ''
+  }
+
+  const rawQuery = Array.isArray(searchQuery) ? searchQuery[0] : searchQuery
```

Comment:
Nit: If `rawQuery` is already `const rawQuery = searchParams[SEARCH_KEY]` - Then maybe this can be `rawQueryValue`?